### PR TITLE
add a utility script to run rebench with a configurable rir repo

### PR DESCRIPTION
**Pull request by o-**
how about somthing like this?

Usage is a simple:

```
run.sh rirVmDir gnuVmDir [--warmup N] [other rebench options]
```

works for any kind of checkout and path. We could use it from the gitlab ci instead of the tools scripts and we could use it on the benchmark machine (i didn't touch this part, since I don't understand the setup dir).

all the benchmark paths are relative and all the vm paths are passed as direct options to run.sh

### CODE REVIEW

**Comment by o-** [comment]
see https://github.com/reactorlabs/rir/pull/432 for how this would simplify the gitlab ci scripts

File: run.sh [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+set -e
+
+SCRIPTPATH=`cd $(dirname "$0") && pwd`
+if [ ! -d $SCRIPT_PATH ]; then
+  echo "Could not determine absolute dir of $0"
+  echo "Maybe accessed with symlink"
+fi
+BENCHMARKS="$SCRIPTPATH"
+
+RIR_VM=$(readlink -f $1)
```
Comment (charig) [inline run.sh]: This does not work in MacOS. Use this instead if OS != linux
https://github.com/reactorlabs/rir/blob/master/tools/downloadBenchs.sh#L53

File: template/rebench.conf [hunk 2]
```diff
@@ -0,0 +1,300 @@
+# Config file for ReBench
```
Comment (charig) [inline template/rebench.conf]: Why the template?

**Comment by charig** [comment]
So, as I said, IMO the thing was not so complex nor it deserved a cleanup. Now.. the script is here it is probably worth integrating it.

However, there are a couple of things missing in order to make it usable for both use cases:

1) The previous --travis option added some options to the rebench command so that it does not send the results to the web server, etc. 

2) IMO it is not necessary to have a template. The repo using this script should provide a proper rebench.conf. For the RBenchmarking the rebench.conf is there. For RIR we are downloading the rebench.conf using `downloadBenchs.sh`. So, if we are removing code duplication I'd not add duplication with the rebench template in RIR that is not necessary. Maybe if you want to make it more flexible just add another parameter to the script with the location of the rebench.conf.

**Comment by o-** [comment]
> The previous --travis option added some options to the rebench command so that it does not send the results to the web server, etc.

this is taken care off, the run script passes all remaining parameters to rebench. this is how gitlab-ci.yaml is passing -S to it for example.

> IMO it is not necessary to have a template. The repo using this script should provide a proper 
> rebench.conf. For the RBenchmarking the rebench.conf is there.

I guess this is my main disagreement. if we need to generate a rebench config, because most parameters cannot be supplied to rebench at the cli, then we should do it cleanly and in every case, so we don't have two ways of doing it.

> For RIR we are downloading the rebench.conf using downloadBenchs.sh.

That would be removed in rir/ #432, not necessary anymore

> So, if we are removing code duplication I'd not add duplication with the rebench template in RIR that is not necessary.

the idea was that after changing the benchmarking machines script we would delete the main rebench.conf

**Comment by o-** [comment]
> So, as I said, IMO the thing was not so complex nor it deserved a cleanup. 

ok, then you can close this

**Comment by o-** [comment]
maybe a highlevel explanation of how this script works:

1. creates a temp directory
2. sets up a benchmark env in that directory:
  * create a rebench config according to template
  * copy all benchmarks into it
3. executes rebench with all remaining arguments passed
4. removes temp directory

my goal was to have an idempotent script that is

1 self contained
2 does not modify any persistent files
3 reproducible

**Comment by o-** [comment]
actually, even nicer would be to do all of this in a docker container...

**Pull request merged**